wolfcrypt/src/debug.c

```diff
@@ -0,0 +1,42 @@
+/* debug.c
+ *
+ * Copyright (C) 2006-2020 wolfSSL Inc.
+ *
+ * This file is part of wolfSSL.
+ *
+ * wolfSSL is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * wolfSSL is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1335, USA
+ */
+
+
+#ifdef HAVE_CONFIG_H
+    #include <config.h>
+#endif
+
+#ifndef WOLFSSL_USER_SETTINGS
+    #include <wolfssl/options.h>
```

Comment:
Doesn't WOLFSSL_USER_SETTINGS control include of "user_settings.h" which is done by settings.h already?  options.h is not available on all platforms and shouldn't be included in wolfSSL proper.

Reply:
Fixed.  The "options.h" include came from copying over `wolfcrypt/test/test.c` -- there's some abstraction barrier violations in there alas, e.g. for `HAVE_STACK_SIZE` it includes ssl.h.  But I was able to eliminate the violations completely for the new debug.c.  Gonna go add build gating now.